automation/repeated_test.sh

```diff
@@ -44,6 +44,28 @@ function new_tests {
         | sed -E 's/\|$//'
 }
 
+# taking into account that each file containing changed tests is run several times per default
+# and considering overhead of cluster-up etc., we should just skip the run
+# if the total number of tests for all runs gets higher than the total number of tests
+function should_skip_test_run_due_to_too_many_tests() {
+    local new_tests="$1"
+    local test_start_pattern='(Specify|It|Entry)\('
+    local tests_total_estimate=0
+    while IFS= read -r -d '' test_file_name; do
+        tests_total_estimate=$(( tests_total_estimate + $(grep -hcE "${test_start_pattern}" "$test_file_name") ))
```

Comment:
$ find tests/ -name '*.go'  | xargs grep It | wc -l
765
$ find tests/ -name '*.go'  | xargs grep "It(" | wc -l 
513 

Please consider adding "(" so the count will be more accurate
we can see it affects